pennylane/resource/resource.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The data class which will aggregate all the resource information from a quantum workflow.
+"""
+
+
+class Resources:
+    r"""Contains attributes which store key resources such as number of gates, number of wires, shots,
+    depth and gate types, tracked over a quantum workflow.
```

Comment:
```suggestion
    depth and gate types.
```
Minor, but since this isn't always for tracking a whole workflow.